pkg/ccl/backupccl/restore_job.go

```diff
@@ -382,50 +381,61 @@ func restore(
 
 	generativeCheckpointLoop := func(ctx context.Context) error {
 		defer close(requestFinishedCh)
-		for progress := range progCh {
-			mu.Lock()
-			var progDetails backuppb.RestoreProgress
-			if err := pbtypes.UnmarshalAny(&progress.ProgressDetails, &progDetails); err != nil {
-				log.Errorf(ctx, "unable to unmarshal restore progress details: %+v", err)
-			}
+		for {
+			select {
+			case <-ctx.Done():
+				return ctx.Err()
+			case progress, ok := <-progCh:
+				if !ok {
+					return nil
+				}
```

Comment:
Was this update needed for the bug fix as well?

My only concern here is that previously this goroutine exited by virtue of the writer calling close when it was done. Now, if the context gets cancelled nothing is reading from progCh.  We write to progCh in the push meta call back without checking any context there either.  I have no proof that this is a problem, it just smells like a potential area for a problem so might be good to double check.

Reply:
This change wasn't required for the fix, I think I incorrectly convinced myself that this was another place we had overlooked a ctx cancellation check. I'm convinced by your comment so I'll revert this diff, thanks!